core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -50,9 +52,16 @@ object Scheduler {
   type StateOutput = (StateSnapshot, Source[StateEvent, Any])
 
   def fromClient(client: MesosClient): Flow[SpecInput, StateOutput, NotUsed] = {
+    if (!isMultiRoleFramework(client.frameworkInfo)) {
+      throw new IllegalArgumentException(
+        "Mesos client supports only MULTI_ROLE frameworks. Please provide create MesosClient with FrameworkInfo that has capability MULTI_ROLE")
```

Comment:
It's rather: `USI scheduler provides support for MULTI_ROLE frameworks only....`